### PR TITLE
speed up metadata dumping and loading for Tikv 

### PR DESCRIPTION
**Pull request by zhijian-pro**
speed up metadata dumping and loading for Tikv

### CODE REVIEW

File: pkg/meta/tkv.go [hunk 1]
```diff
@@ -2694,10 +2877,41 @@ func (m *kvMeta) LoadMeta(r io.Reader) error {
 		NextSession: 1,
 	}
 	refs := make(map[string]int64)
+
+	//TODO: try raw kv   cli.BatchPut()
```
Comment (davies) [inline pkg/meta/tkv.go:2881]: rawkv is not compatible to transactional kv, we can't use that here.

File: pkg/meta/tkv.go [hunk 2]
```diff
@@ -2598,7 +2756,32 @@ func (m *kvMeta) DumpMeta(w io.Writer) error {
...
+	if _, err = bw.WriteString("\n}\n"); err != nil {
+		return err
+	}
+	if bar.Current() != total {
+		logger.Warnf("Dumped %d / total %d, some entries are not dumped", bar.Current(), total)
+	}
+	bar.SetTotal(0, true)
+	progress.Wait()
+
+	return bw.Flush()
```
Comment (davies) [inline pkg/meta/tkv.go:2784]: We should not fail the dump even bw.Flush() failed

Comment (zhijian-pro) [reply pkg/meta/tkv.go:2784]: Doesn't flush failure mean that the result is incomplete?

Comment (davies) [reply pkg/meta/tkv.go:2784]: my bad, bw is the writer, never mind.

**Comment by davies** [comment]
in favor #998

**Pull request closed without merging**